### DISCUSSION TITLE
Structure repo pour projet ML Ops

### DISCUSSION THREAD

**Discussion by mis7ern9o** [DstMlOpsCrypto/MainCrypto#13] [General]
Avec les différentes lectures effectuées, voici un template de structures de repo pour un projet ML Ops standard qui viendrait compléter ce que tu as déjà fait @FredericLAGNIEZ 

mlops-crypto-prediction/
├── data/
│   ├── raw/                # Données brutes non traitées
│   ├── processed/          # Données traitées et prêtes pour l'analyse
│   └── external/           # Données provenant de sources externes
├── notebooks/
│   ├── exploration/        # Notebooks pour l'analyse exploratoire des données
│   └── modeling/           # Notebooks pour la modélisation et l'entraînement
├── src/
│   ├── data/               # Scripts pour la collecte et la préparation des données
│   ├── models/             # Scripts pour la définition et l'entraînement des modèles
│   ├── evaluation/         # Scripts pour l'évaluation des modèles
│   └── deployment/         # Scripts pour le déploiement des modèles
├── tests/
│   ├── unit/               # Tests unitaires pour les scripts
│   └── integration/        # Tests d'intégration pour les pipelines
├── config/
│   ├── config.yaml         # Fichiers de configuration pour les paramètres du projet
│   └── logging.yaml        # Configuration pour la journalisation
├── scripts/
│   ├── preprocess.py       # Script pour la préparation des données
│   ├── train.py            # Script pour l'entraînement des modèles
│   ├── evaluate.py         # Script pour l'évaluation des modèles
│   └── deploy.py           # Script pour le déploiement des modèles
├── .github/
│   ├── workflows/          # Workflows GitHub Actions pour CI/CD
│   └── ISSUE_TEMPLATE.md   # Modèle pour les issues
├── Dockerfile              # Fichier Docker pour la création de l'image du projet
├── requirements.txt        # Liste des dépendances du projet
├── README.md               # Documentation du projet
└── LICENSE                 # Licence du projet

A disposition pour en discuter

**Comment by TWR-RWT** [comment]
ça me paraît bien. Il y a juste la partie documentation qui devrait peut être apparaître dans ton arborescence ?

**Reply by mis7ern9o** [reply]
Yes 
Concernant la doc : nous avons le README.md à la racine du repo "[MainCrypto](https://github.com/DstMlOpsCrypto/MainCrypto)" concernant le how to expliquer comment deployer la solution, le wiki (https://github.com/DstMlOpsCrypto/MainCrypto/wiki)  utilisé pour la documentation (modeles, pipeline...) et les contributeurs au projet , et un github page (https://dstmlopscrypto.github.io/MainCrypto/)  pour hoster la document utilisateurs - hors equipe de dev et pour communiquer vers l'exterieur car cette page est public

**Comment by mis7ern9o** [comment]
@FredericLAGNIEZ, je vais debuter les tests sur les données + code . je pars sur l'existant et je complete avec cette proposition ?

**Reply by mis7ern9o** [reply]
puisque que nous sommes dans la partie structuration du repo, concernant FastAPI je propose la structure suivante cible que l'on pourra modifier : 
A la racine du repo, creation d'un repertoire fastapi_app :  

├── fastapi_app/
│   ├── auth/
│   │   ├── router.py       # Authentication routes
│   │   ├── schemas.py      # Pydantic models for auth
│   │   ├── models.py       # Database models for auth
│   │   ├── dependencies.py # Dependency injections for auth
│   │   ├── config.py       # Local configs for auth
│   │   ├── constants.py    # Constants for auth
│   │   ├── exceptions.py   # Custom exceptions for auth
│   │   ├── service.py      # Business logic for auth
│   │   └── utils.py        # Utility functions for auth
│   ├── ml/
│   │   ├── router.py       # ML routes
│   │   ├── schemas.py      # Pydantic models for ML
│   │   ├── models.py       # Database models for ML
│   │   ├── dependencies.py # Dependency injections for ML
│   │   ├── config.py       # Local configs for ML
│   │   ├── constants.py    # Constants for ML
│   │   ├── exceptions.py   # Custom exceptions for ML
│   │   ├── service.py      # Business logic for ML
│   │   └── utils.py        # Utility functions for ML
│   ├── config.py           # Global configs
│   ├── models.py           # Global models
│   ├── exceptions.py       # Global exceptions
│   ├── database.py         # Database connection
│   └── main.py             # Entry point for FastAPI app

Quand pensez vous ?

**Reply by FredericLAGNIEZ** [reply]
RAS pour moi à ce stade sur le repo fastAPI, je modifie tout de suite le repo reprenant nos échanges

**Reply by FredericLAGNIEZ** [reply]
Je vois que le repo a déjà été modifié, je te laisse le compléter @mis7ern9o

**Comment by mis7ern9o** [comment]
autre chose, pour ML Flow il faudrait a priori interger un fichier de tracking dans le repertoire script (je fais la formation ML Flow en ce moment), j'en serais un peu plus ds qlq hrs ou jours ;-) )

├── scripts/
│   └── (...)
│   └── mlflow_tracking.py # Script for MLflow tracking

**Comment by mis7ern9o** [comment]
hello, j'ai avancé sur le deploiement avec l'infra à notre disposition (3  EC2).

- utilisation d'un seul repo (cela a dejà été acté)
- 1 repertoire par application (fastapi, mlflow, data-integration)
- pour la base de donnée, nous n'allons pas la reinstallé à chaque Pull Request. Les données brutes seront stockées sur la VM de la base de donnée. Donc un repertoire dédié doit être créé

Ce qui nous ferait : 

├── maincrypto
│ ├── fastapi/
│ ├── mlflow/
│ ├── data-integration/
│ ├── db/

Qu'en pensez vous ?

Pour le deploiement rendez vous sur la [discussion](https://github.com/DstMlOpsCrypto/MainCrypto/discussions/26)
